litmus-portal/graphql-server/pkg/chaos-workflow/ops/operations.go

```diff
@@ -329,3 +371,41 @@ func SendWorkflowEvent(wfRun model.WorkflowRun, r *store.StateData) {
 	}
 	r.Mutex.Unlock()
 }
+
+// ResiliencyScoreCalculator calculates the Rscore and returns the execdata string
+func ResiliencyScoreCalculator(execData string, wfid string) string {
+	var resiliency_score, weightSum, totalTestResult, totalExperiments, totalExperimentsPassed int = 0, 0, 0, 0, 0
+	var jsonData WorkflowEvent
+	json.Unmarshal([]byte(execData), &jsonData)
+	for _, value := range jsonData.Nodes {
+		if value.Type == "ChaosEngine" {
+			if value.ChaosExp == nil {
+				continue
+			}
+			chaosWorkflows, _ := dbOperationsWorkflow.GetWorkflows(bson.D{{"workflow_id", bson.M{"$in": []string{wfid}}}})
+			for _, workflow := range chaosWorkflows {
+				var Weightages []*model.Weightages
+				copier.Copy(&Weightages, &workflow.Weightages)
+				totalExperiments = len(Weightages)
+				for _, weightEntry := range Weightages {
+					if weightEntry.ExperimentName == value.ChaosExp.ExperimentName {
+						x, _ := strconv.Atoi(value.ChaosExp.ProbeSuccessPercentage)
+						totalTestResult += weightEntry.Weightage * x
+						weightSum += weightEntry.Weightage
+					}
+					if value.ChaosExp.ExperimentVerdict == "Pass" {
+						totalExperimentsPassed += 1
+					}
+				}
```

Comment:
This loop can be moved out too, you can create a map of the probe success % and then outside the workflow node loop run the weightage loop and find the result required by mapping the exp names